packages/babel-helper-replace-supers/src/index.js

```diff
@@ -188,139 +176,143 @@ export default class ReplaceSupers {
-        ]),
-        t.expressionStatement(
-          t.assignmentExpression(
-            "=",
-            node.left,
-            t.binaryExpression(
-              node.operator.slice(0, -1),
-              t.cloneNode(ref),
-              node.right,
-            ),
-          ),
-        ),
-      ];
+    // super.test()
+    // though, it's SUPER.prototype.test() after the above.
+    const grandParentPath = parentPath.parentPath;
+    const callee = parentPath.node;
+    if (grandParentPath.isCallExpression({ callee })) {
+      grandParentPath
+        .get("callee")
+        .replaceWith(t.memberExpression(callee, t.identifier("call")));
+      grandParentPath.node.arguments.unshift(t.thisExpression());
     }
   }
 
-  specHandle(path: NodePath) {
-    let property;
-    let computed;
-    let args;
-
-    const parent = path.parent;
-    const node = path.node;
-
-    if (isIllegalBareSuper(node, parent)) {
-      throw path.buildCodeFrameError("Illegal use of bare super");
+  specHandleAssignmentExpression(path) {
+    const { node } = path;
+    const { operator } = node;
+    const { computed, property } = node.left;
+    if (operator === "=") {
+      // super.name = "val"
+      // to
+      // _set(Object.getPrototypeOf(objectRef.prototype), "name", this);
+      const setter = this.setSuperProperty(property, node.right, computed);
+      return [setter];
     }
 
-    if (t.isCallExpression(node)) {
-      const callee = node.callee;
-      if (t.isSuper(callee)) {
-        return;
-      } else if (isMemberExpressionSuper(callee)) {
-        // super.test();
-        // to
-        // _get(Object.getPrototypeOf(objectRef.prototype), "test", this).call(this);
-        property = callee.property;
-        computed = callee.computed;
-        args = node.arguments;
-      }
-    } else if (t.isMemberExpression(node) && t.isSuper(node.object)) {
-      // super.name;
-      // to
-      // _get(Object.getPrototypeOf(objectRef.prototype), "name", this);
-      property = node.property;
-      computed = node.computed;
-    } else if (
-      t.isUpdateExpression(node) &&
-      isMemberExpressionSuper(node.argument)
-    ) {
-      const binary = t.assignmentExpression(
-        node.operator[0] + "=",
-        node.argument,
+    // super.age += 2;
+    // to
+    // _set(
+    //   Object.getPrototypeOf(objectRef.prototype),
+    //   "name",
+    //   _get(Object.getPrototypeOf(CLASS.prototype), "METHOD", this) + 2,
+    //   this,
+    //   true,
+    // );
+    // TODO this needs cleanup. Should be a single proto lookup
+    const ref = path.scope.generateDeclaredUidIdentifier("ref");
+    const setter = this.setSuperProperty(
+      property,
+      t.binaryExpression(operator.slice(0, -1), t.cloneNode(ref), node.right),
+      computed,
+    );
+    return [
+      t.assignmentExpression(
+        "=",
+        t.cloneNode(ref),
+        this.getSuperProperty(property, computed),
+      ),
+      setter,
```

Comment:
Why do we need to store the `_get` result?

Reply:
We don't yet have a set algorithm that will do the get, too. So, we have to do the get, then pass it to the set.

And, given the codesize for the get/set algorithms, I didn't want in include yet another for this case.